repos/system_upgrade/common/models/repositoriesmap.py

```diff
@@ -2,34 +2,93 @@
-    """Target RHEL repoid as present in the Red Hat CDN"""
-    to_pes_repo = fields.String()
-    """Target RHEL repo name as used in the Package Evolution Service database"""
-    from_minor_version = fields.String()
-    """To which source RHEL minor versions the mapping relates to"""
-    to_minor_version = fields.String()
-    """To which target RHEL minor versions the mapping relates to"""
-    arch = fields.String()
-    """CPU architecture the mapping relates to"""
-    repo_type = fields.StringEnum(choices=['rpm', 'srpm', 'debuginfo'])
-    """Type of content the mapped repos hold"""
-
-
-class RepositoriesMap(Model):
+    pesid = fields.String()
+    """
+    The PES id of the repository.
+
+    The PES id indicate family of YUM repositories. E.g. rhel8-BaseOS covers
+    variants of BaseOS YUM repositories for all channels, architectures,
+    RHUI, etc, which have basically same purpose.
+    """
+
+    major_version = fields.String()
+    """
+    The major version of OS.
+
+    E.g. for RHEL 7.9 the major version is 7. Since we work with versions
+    as with strings throughout the whole codebase, we keep this data type here too.
+    """
+
+    repoid = fields.String()
+    """
+    The repository ID which identifies the repository from YUM/DNF POV.
+    """
+
+    arch = fields.StringEnum(['x86_64', 's390x', 'ppc64le', 'aarch64'])
+    """
+    The architecture for which the repository is delivered.
+    """
+
+    repo_type = fields.StringEnum(['rpm', 'debug', 'srpm'])
+    """
+    The repository type.
+
+    In our case, we usually map just repositories with the 'rpm' type, so
+    usually you can see just this one, but the others are possible to add
+    too.
+    """
+
+    channel = fields.StringEnum(['ga', 'tuv', 'e4s', 'eus', 'aus', 'beta'])
+    """
+    The 'channel' of the repository.
+
+    The 'channel' could be a little bit inaccurate term, but let's use it in
+    this project. The standard repositories has 'ga' channel. 'beta'
+    repositories are unsupported for IPU, however they are useful for testing
+    purposes. The other channels indicate premium repositories.
+    """
+
+    rhui = fields.StringEnum(['', 'aws', 'azure'])
+    """
+    Indicate whether the repository is deliver for RHUI and which one.
+
+    For non-rhui systems: empty string
+    For AWS or Azure: 'aws' / 'azure'
+    """
+
+
+class RepositoriesMapping(Model):
+    """
+    Private model containing information about mapping between repositories.
+
+    Warning: We expect to be only consumers of this model.
+    This means the model is not covered by deprecation process and can be
+    changed or removed any time.
+    """
     topic = TransactionTopic
 
-    repositories = fields.List(fields.Model(RepositoryMap), default=[])
-    """List of source RHEL repo <-> target RHEL repo mappings"""
+    mapping = fields.List(fields.Model(RepoMapEntry))
+    repositories = fields.List(fields.Model(PESIDRepositoryEntry))
```

Comment:
```suggestion
    mapping = fields.List(fields.Model(RepoMapEntry), default=[])
    repositories = fields.List(fields.Model(PESIDRepositoryEntry), default=[])
```
For default initialization to fill the fields. Makes it easier to handle the next() blah cases

Reply:
I wanted originally skip the default here, as we expect the message will always contain non-empty lists. But in the end, we possibly drop the model in future as well. So let's do it. (done)